MedscribeUI/src/components/Utilities/ThreeBtnSelector.tsx

```diff
@@ -0,0 +1,33 @@
+import { Button } from '@mantine/core';
+import { useState, useEffect } from 'react';
+
+interface ThreeBtnSelectorProps {
+  buttonLabelOptions: string[];
+  initialSelected?: string;
+  onSelect: (value: string) => void;
+}
+
+function ThreeBtnSelector({ buttonLabelOptions, initialSelected = '', onSelect }: ThreeBtnSelectorProps) {
+  const [selected, setSelected] = useState(initialSelected);
+
+  useEffect(() => {
+    setSelected(initialSelected);
+  }, [initialSelected]);
+
+  const handleSelect = (value: string) => {
+    setSelected(value);
+    onSelect(value);
```

Comment:
we don't want to fire any event until the regen button is clicked. All this does is update the state. We should remove this. the layout will be responsible for listening when the regen button is clicked and then firing off the api call with the new note control values.

Reply:
Ignore the above comment sorry. Instead can you remove this function entirely. We absolutley do not want this tiny unit component to have its own state. We want to pass it in from where we centralize our components and that is the layout.